src/PrestaShopBundle/Bridge/Controller/ControllerConfigurationFactory.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShopBundle\Bridge\Controller;
+
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorage;
+use Tools;
+
+/**
+ * Create configuration for controller migrate horizontally
+ */
+class ControllerConfigurationFactory
+{
+    /**
+     * @var TokenStorage
+     */
+    private $tokenStorage;
+
+    public function __construct(
+        TokenStorage $tokenStorage
+    ) {
+        $this->tokenStorage = $tokenStorage;
+    }
+
+    public function create(array $configuration = []): ControllerConfiguration
```

Comment:
Please use [array shapes](https://phpstan.org/writing-php-code/phpdoc-types#array-shapes) to document the allowed properties.

That said, I'm not a big fan of using open-ended structures like for parameters. Why not pass the required parameters as individual parameters instead?